packages/commonwealth/client/scripts/controllers/app/webWallets/keplr_web_wallet.ts

```diff
@@ -48,37 +47,30 @@ class KeplrWebWalletController implements IWebWallet<AccountData> {
     return this._offlineSigner;
   }
 
-  public async signLoginToken(
-    message: string,
-    address: string
-  ): Promise<StdSignature> {
-    const signature = await window.keplr.signArbitrary(
-      this._chainId,
-      address,
-      message
-    );
-    return signature;
+  getChainId() {
+    return this._chainId;
   }
 
-  public async signWithAccount(account: Account): Promise<string> {
-    const webWalletSignature = await this.signLoginToken(
-      account.validationToken.trim(),
-      account.address
-    );
-    const signature = {
-      signature: {
-        pub_key: webWalletSignature.pub_key,
-        signature: webWalletSignature.signature,
-      },
+  public async getRecentBlock(chainIdentifier: string) {
+    const url = `${window.location.origin}/cosmosAPI/${
```

Comment:
Also use injected provider here, ideally. We may want to specify osmosis specifically.

Reply:
So the Keplr injected provider (I assume by this you mean `window.keplr`) doesn't let you request the latest block, you have to use the Stargate client as I have done here.